rest/ProductService.js

```diff
@@ -82,11 +88,21 @@ class ProductService {
             throw new IllegalArgumentError(INVALID_QUANTITY_ERR);
         }
     }
+    
 
-    /**
-     * TODO
-     * Validate serial numbers
-     */
+    #assertProductIsActive(productId) {
+        const product = this.#productPersistence.getById(productId);
+        if (!product.isActive) {
+            throw new Error(INVALID_IS_ACTIVE_ERR);
+        }
+    }
+
+    #assertNoAssociatedOrders(productId) {
+        const orders = this.#orderPersistence.getByProductId(productId);
+        if (orders.length > 0) {
```

Comment:
if(!this.#orderPersistence.getByProductId(productId))

You can just check if the array has a truthy value (other than 0/undefined in that case)